core/src/main/java/hudson/model/Job.java

```diff
@@ -475,27 +476,75 @@ public boolean supportsLogRotator() {
         return true;
     }
 
+
+    /**
+     * Search the build by build number. This function is for search box which is at the nav bar
+     *
+     * @param token The keyword entered in the search box
+     * @param result Result list that will show on the page
+     */
+    void searchByBuildNumber(String token, List<SearchItem> result) {
+        token = token.substring(1); // ignore leading '#'
+        try {
+            int n = Integer.parseInt(token);
+            Run b = getBuildByNumber(n);
+            if (b == null)
+                return; // no such build
+            result.add(SearchItems.create("#" + n, "" + n, b));
+        } catch (NumberFormatException e) {
+            // not a number
+        }
+    }
+
     @Override
     protected SearchIndexBuilder makeSearchIndex() {
         return super.makeSearchIndex().add(new SearchIndex() {
             @Override
             public void find(String token, List<SearchItem> result) {
-                try {
-                    if (token.startsWith("#"))
-                        token = token.substring(1); // ignore leading '#'
-                    int n = Integer.parseInt(token);
-                    Run b = getBuildByNumber(n);
-                    if (b == null)
-                        return; // no such build
-                    result.add(SearchItems.create("#" + n, "" + n, b));
-                } catch (NumberFormatException e) {
-                    // not a number.
+
+                if (token.startsWith("#")) { // Search by build number
+                    searchByBuildNumber(token, result);
+                } else { // Search by build display name
+                    boolean isCaseSensitive = UserSearchProperty.isCaseInsensitive();
```

Comment:
Because code in `filter` lambda are different. Therefore, I didn't extract them.

The reason I extract `searchByBuildNumber`  is because they are the same. Thus, I deduplicate them.

Would it also be a good idea to extract them?